src/platforms/android/androidutils.cpp

```diff
@@ -134,3 +135,73 @@ void AndroidUtils::appReviewRequested() {
       "org/mozilla/firefox/vpn/qt/VPNAppReview", "appReviewRequested",
       "(Landroid/app/Activity;)V", activity.object<jobject>());
 }
+
+// static
+void AndroidUtils::dispatchToMainThread(std::function<void()> callback) {
+  QTimer* timer = new QTimer();
+  timer->moveToThread(qApp->thread());
+  timer->setSingleShot(true);
+  QObject::connect(timer, &QTimer::timeout, [=]() {
+    callback();
+    timer->deleteLater();
+  });
+  QMetaObject::invokeMethod(timer, "start", Qt::QueuedConnection);
+}
+
+// static
+QByteArray AndroidUtils::getQByteArrayFromJString(JNIEnv* env, jstring data) {
+  const char* buffer = env->GetStringUTFChars(data, nullptr);
+  if (!buffer) {
+    logger.error() << "getQByteArrayFromJString - failed to parse data.";
+    return QByteArray();
+  }
+  QByteArray out(buffer);
+  env->ReleaseStringUTFChars(data, buffer);
+  return out;
+}
+
+// static
+QString AndroidUtils::getQStringFromJString(JNIEnv* env, jstring data) {
+  const char* buffer = env->GetStringUTFChars(data, nullptr);
+  if (!buffer) {
+    logger.error() << "getQStringFromJString - failed to parse data.";
+    return QString();
+  }
+  QString out(buffer);
+  env->ReleaseStringUTFChars(data, buffer);
+  return out;
+}
+
+// static
+QJsonObject AndroidUtils::getQJsonObjectFromJString(JNIEnv* env, jstring data) {
+  QJsonObject empty = QJsonObject();
+
+  const char* buffer = env->GetStringUTFChars(data, nullptr);
+  if (!buffer) {
+    logger.error() << "getQJsonObjectFromJString - failed to parse data.";
+    return empty;
+  }
+  QByteArray raw(buffer);
+  env->ReleaseStringUTFChars(data, buffer);
+
+  logger.debug() << "getQJsonObjectFromJString - parsing raw data: "
```

Comment:
What about this:
1. No 'raw data' logging here.
2. if the caller of this method receives an empty object, it can log: `"fail to parse string:" << getQStringFromJString(env, data);`

Reply:
I don't think that is even necessary. If we get in the if statements for errors we log so there's no logging data just for the hell of it.